### ISSUE TITLE
29 -> 30 database migration panic

### ISSUE THREAD

**Issue by marcelo-gonzalez** [near/nearcore#5936]
I got this error running neard built from the current HEAD (42e4e129b37a8462d6045ec07f8f):

```bash
marcelo@marcelo-debian:~/nearcore$ ./target/release/neard --verbose debug run
Dec 21 15:43:30.877  INFO neard: Version: trunk, Build: crates-0.10.0-312-g42e4e129b, Latest Protocol: 50
Dec 21 15:43:30.884  INFO near: Opening store database at "/home/marcelo/.near/data"
Dec 21 15:43:30.897  INFO near: Migrate DB from version 29 to 30
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: InvalidData, error: "Not all bytes read" }', core/store/src/migrations.rs:791:59
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Now, I had previously stopped (with ctrl-C) neard while it was doing this migration, so something must have been left in a bad state. Is stopping neard mid-migration supposed to supported as a safe operation? If not, would maybe make sense to print a message saying not to stop the process until it's finished.

Currently trying to look more at what the state of the chain is, but it's a little hard since I can't use state viewer, probably for the same reason.

**Comment by vorotech** [comment]
Got the same issue. Has stopped with Ctrl+C after it seemed to be stuck with last entry in the log `INFO near: Migrate DB from version 29 to 30`
Next time I was trying to launch it it was giving the same error. Currently trying to find a way to fix the broken state.

**Comment by bowenwang1996** [comment]
> Now, I had previously stopped (with ctrl-C) neard while it was doing this migration, so something must have been left in a bad state. Is stopping neard mid-migration supposed to supported as a safe operation? If not, would maybe make sense to print a message saying not to stop the process until it's finished.

Yeah that is the problem. I believe the migrations are not written in an idempotent way, especially if they involve changes to some data structure.

**Comment by bowenwang1996** [comment]
Should be addressed with #6282
